### ISSUE TITLE
MIDI learn Split Key

### ISSUE THREAD

**Issue by mkruselj** [surge-synthesizer/surge#624]
![image](https://user-images.githubusercontent.com/2393720/52953849-f2960e80-3388-11e9-9816-7def8caff89c.png)

**Comment by mkruselj** [comment]
Instead of Ctrl+click, for consistency it's probably better to have the MIDI learn in the RMB context menu, right below Edit Value.

**Comment by baconpaul** [comment]
OK here's how it works now

The 'learn' function in the RMB sets synth->learn_param to the param_id you want to learn.  SGEVC around line 206. That's buried in createMidiLearnBlahDeBlah but for the split key you can just add a special menu callback which adds the learn set to the param id also. Fine. You can see how that code works.

Once learn_param is set the synth checks the value in channel_controller and sets up the data structure. We have it a bit easier.

In playNote you can also check

```
if (learn_param >= 0 && params[learn_param].type == ct_splitwhatzit)'
    learn_param = -1;
    split_value = note
```

basically. Do those two things and you are good. You should also in channel_controller suppress learn if params[learn_param].type == ct_splitwhatzit. This is probably best done by adding a new method to Parameter

so `bool Parameter::learnByNoteRatherThanController() { return type == ct_splitwhatzit; }` then check that in the two spots (once check it positive once check it negative).

That should get you most of the way there. Some things like show the midi icon also are in SGEVC and are pretty clear.
